### PR TITLE
Split up concerns and use ArgParse instead of getopt

### PR DESCRIPTION
**Pull request by InblEric**

### CODE REVIEW

File: analyze.py [hunk 1]
```diff
@@ -1,163 +1,19 @@
...
+from graph_listing import graph_listing
 from utils import time_func
 
 
-plt.style.use('seaborn-whitegrid')
+parser = argparse.ArgumentParser(description="Analyze a TopShot Listing")
+parser.add_argument("-u", "--url", metavar='u', type=str, help="URL of a moment (required)")
+parser.add_argument("-m", "--maxprice", type=int, help="Enter a maximum price you are willing to spend", required=False)
+parser.add_argument("-a", "--all", action='store_const', const=True, default=False, help="Enter a maximum price you are willing to spend", required=False)
+parser.add_argument("-g", "--graph", action='store_const', const=True, default=False, help="Enter a maximum price you are willing to spend", required=False)
```
Comment (MichaelNotEric) [inline analyze.py]: 11 and 12 should have different help messages right?

File: analyze_listing.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import csv
+
+from listings import get_all_listings_from_url
+
+def analyze__and_display_listing(url, max_price, all_listings, listings):
```
Comment (MichaelNotEric) [inline analyze_listing.py]: why the double underscore?

**Review by MichaelNotEric — CHANGES_REQUESTED** [review]

File: analyze.py (hunk 1 above)
Comment (InblEric) [reply analyze.py]: Yes

File: analyze_listing.py (hunk 2 above)
Comment (InblEric) [reply analyze_listing.py]: Because I type bad

File: analyze.py [hunk 3]
```diff
@@ -1,163 +1,19 @@
...
-import getopt
-import json
-import matplotlib.pyplot as plt
-import numpy as np
-import requests
-import sys
+import argparse
 
-from bs4 import BeautifulSoup
+from analyze_listing import analyze__and_display_listing
```
Comment (MichaelNotEric) [inline analyze.py]: needs to be changed here too

File: analyze.py [hunk 4]
```diff
@@ -1,163 +1,19 @@
...
+from graph_listing import graph_listing
 from utils import time_func
 
 
-plt.style.use('seaborn-whitegrid')
+parser = argparse.ArgumentParser(description="Analyze a TopShot Listing")
+parser.add_argument("-u", "--url", metavar='u', type=str, help="URL of a moment (required)")
+parser.add_argument("-m", "--maxprice", type=int, help="Enter a maximum price you are willing to spend", required=False)
+parser.add_argument("-a", "--all", action='store_const', const=True, default=False, help="Enter a maximum price you are willing to spend", required=False)
```
Comment (MichaelNotEric) [inline analyze.py]: this should be something like "Use this flag if you want to display all listings, ignoring value

**Review by MichaelNotEric — CHANGES_REQUESTED** [review]

File: analyze.py (hunk 3 above)
Comment (InblEric) [reply analyze.py]: oops

File: analyze.py (hunk 4 above)
Comment (InblEric) [reply analyze.py]: Wow lol I can't read

**Review by MichaelNotEric — APPROVED** [review]
LGTM

**Pull request merged**